### ISSUE TITLE
Remote connection to Agent freezing

### ISSUE THREAD

**Issue by AdminisTazzer** [Ylianst/MeshCentral#3155]
Hi,

It's been about a week since I noticed this issue, when I am connected to an Agent, the connection freezes and I have to disconnect/reconnect for it to work again. When I do this, it shows as if my previous connection had not ended, it shows as if I had 2 or more sessions (one for each time I disconnect/reconnect) and it keeps doing this over and over... some times it takes about 30sec to freeze, sometimes I get about 2 minutes before it happens. The Server and Agents on which I am experiencing this issue are all up to date (agent:Aug 30 2021 119140246, Server:0.9.28). After a couple of disconnect/reconnect sometimes the agent looks like it crashes. (it goes offline for a moment and comes back up, and on a couple of occasions, it didn't come back up and I had to restart the service.) I've had this problem on workstations and servers, all on windows OS.

As far as I know we haven't changed anything in our environment prior to when this issue started to happen.

In the device console I get this error when the crash happens:

uncaughtException1: Error:  => EventEmitter.emit(): Event dispatch for 'end' on 'http.WebSocketStream.decoded' threw an exception: TypeError: invalid base value in method 'onTunnelClosed()'

Regards,

**Comment by krayon007** [comment]
Are you actively using the remote desktop when it freezes, or is it after it is sitting idle? You shouldn't be getting that exception, so I'll see if I can reproduce that, but it looks unrelated to what you are experiencing. It looks like the tunnel is getting closed. Are you by change behind a proxy or reverse proxy?  The only things off the top of my head that would cause a tunnel disconnect, is the user closing the user consent bar and an idle timeout on the connection, either on the proxy side or the agent side.

**Comment by krayon007** [comment]
I just noticed you said that the previous connection still shows as being connected... This is starting to look like a proxy is disconnecting the tunnel. I Think the connection showing that it is still there might be related to that exception when the agent was trying to clean up.

**Comment by AdminisTazzer** [comment]
Yes it is while I am actively using it, it starts off normal then after a moment, the image freezes and I can't interact with it anymore. It happens even on computers that do not have the User consent bar, like servers, which I am the only one on the computer when it happens. I will look into our setup see if it would be a proxy problem.

**Comment by AdminisTazzer** [comment]
Hi I just wanted to let you know that we have found out what was causing our problem. It was our SOPHOS IPS that was detecting an "intrusion attempt SERVER-APACHE Apache Tomcat WebSocket Infinite Loop CVE-2020-13935 Denial of Service". We think that it was caused at the agent level, probably due to a recent update.
